compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -945,12 +945,17 @@ trait Applications extends Compatibility {
       val app1 =
         if (!success || typedArgs.exists(_.tpe.isError)) app0.withType(UnspecifiedErrorType)
         else {
-          if !sameSeq(args, orderedArgs)
-             && !isJavaAnnotConstr(methRef.symbol)
-             && !typedArgs.forall(isSafeArg)
-          then
+          if isJavaAnnotConstr(methRef.symbol) then
+            // #19951 Make sure all arguments are NamedArgs for Java annotations
+            if typedArgs.exists(!_.isInstanceOf[NamedArg]) then
+              typedArgs = typedArgs.lazyZip(methType.asInstanceOf[MethodType].paramNames).map {
+                case (arg: NamedArg, _) => arg
+                case (arg, name)        => NamedArg(name, arg)
```

Comment:
Scala 2 mandates all arguments to Java annotations to be named in source. The only exception is if there's a single positional argument, then `@Ann(x)` expands to `@Ann(value = x)`. This matches what Java requires for using annotations. Would it make sense to do the same in Scala 3?

Reply:
I think it would make sense, yes. However that would be a source-breaking change, which is in a different league compared to this PR.

Reply:
OK, I'll create a ticket.